include/seqan3/range/view/detail.hpp

```diff
@@ -13,301 +13,296 @@
-     * \{
-     */
-    pipable_adaptor_base() = default;
-    constexpr pipable_adaptor_base(pipable_adaptor_base const &) = default;
-    constexpr pipable_adaptor_base(pipable_adaptor_base &&) = default;
-    constexpr pipable_adaptor_base & operator =(pipable_adaptor_base const &) = default;
-    constexpr pipable_adaptor_base & operator =(pipable_adaptor_base &&) = default;
-    ~pipable_adaptor_base() = default;
-    //!\brief befriend the derived type so that it can instantiate
-    friend derived_type;
-    //!\}
+    //!\brief Type of the CRTP-base.
+    using base_type = adaptor_base<combined_adaptor<left_adaptor_t, right_adaptor_t>,
+                                   left_adaptor_t,
+                                   right_adaptor_t>;
+
+    //!\brief Befriend the base class so it can call impl().
+    friend base_type;
+
+    //!\brief Combine all arguments via `operator|`.
+    template <std::ranges::InputRange urng_t,
+              typename left_adaptor_t_,
+              typename right_adaptor_t_>
+    static auto impl(urng_t && urange, left_adaptor_t_ && left_adaptor, right_adaptor_t_ && right_adaptor)
+    {
+        return std::forward<urng_t>(urange)
+             | std::forward<left_adaptor_t_>(left_adaptor)
+             | std::forward<right_adaptor_t_>(right_adaptor);
+    }
+
+public:
+    //!\brief Inherit the base type's constructors.
+    using base_type::base_type;
+
+    //!\brief Store both arguments in the adaptor.
+    combined_adaptor(left_adaptor_t l, right_adaptor_t r) :
+        base_type{std::forward<left_adaptor_t>(l), std::forward<right_adaptor_t>(r)}
+    {}
 };
 
 // ============================================================================
-//  generic_pipable_view_adaptor
+//  adaptor_for_view_without_args
 // ============================================================================
 
-/*!\brief   A class template template that makes a view pipeable.
+/*!\brief Template for range adaptor closure objects that store no arguments and always delegate to the
+ *        view constructor.
+ * \tparam view_type The view type template.
  * \ingroup view
- * \tparam  view_type The view template that your wish to define the adaptor for.
  *
  * \details
  *
- * A full view implementation consists of three entities:
+ * # Example
  *
- *   1. the actual view, e.g. `view_foo`;
- *   2. an adaptor that returns instances of 1. and enables usage in pipes, e.g. `foo_fn`
- *   3. an instance of the adaptor, e.g. `view::foo` that is usable by consumers of your view
+ * From include/seqan3/range/view/single_pass_input.hpp:
  *
- * For more details, see the HowTo on writing views in the SeqAn3 wiki: https://github.com/seqan/seqan3/wiki
+ * \snippet include/seqan3/range/view/single_pass_input.hpp view_def
  *
- * This template generates the adaptor for you:
+ * This is the signature of the view type template in namespace seqan3::detail.
```

Comment:
Ah ok the sentence here belongs to the snippet above? This is confusing because you read it as if 
`This ... :` and now comes this. It is much easier to understand if you put this sentence before the snippet.